handsontable/test/e2e/core/setDataAtCell.spec.js

```diff
@@ -106,4 +106,78 @@ describe('Core.setDataAtCell', () => {
 
     expect(beforeChange).toHaveBeenCalledWith([[0, 1, 'foo', 'foo2']], 'multiple-change');
   });
+
+  it('should keep boolean value in the cell after copy and paste checkbox', () => {
+    const hot = handsontable({
+      data: [
+        ['Tagcat', true],
+        ['Zoomzone', false],
+        ['Meeveo', false],
+        ['Buzzdog', true],
+        ['Katz', true],
+      ],
+      colHeaders: ['Company name', 'In stock'],
+      columns: [
+        { data: 0, type: 'text' },
+        {
+          data: 1,
+          type: 'checkbox',
+          className: 'htCenter',
+        },
+      ],
+    });
+
+    const plugin = hot.getPlugin('CopyPaste');
+    const copyEvent = getClipboardEvent('copy');
+
+    selectCell(0, 1);
+
+    plugin.onCopy(copyEvent);
+
+    selectCell(1, 1);
+
+    expect(getDataAtCell(1, 1)).toBe(false);
+
+    plugin.paste(copyEvent.clipboardData.getData('text/plain'));
+
+    expect(getDataAtCell(1, 1)).toBe(true);
+  });
```

Comment:
I'd keep the test case as simple as possible
```suggestion
  it('should keep boolean value in the cell after copy and paste checkbox', () => {
    const hot = handsontable({
      data: [
        [true],
        [false],
        [false],
      ],
      type: 'checkbox',
    });

    setDataAtCell(0, 0, 'true')

    expect(getDataAtCell(0, 0)).toBe(true);

    setDataAtCell(0, 0, 'false')

    expect(getDataAtCell(0, 0)).toBe(false);
  });
```